### PR TITLE
Add tests for TimeParser

### PR DESCRIPTION
**Pull request by kiwibang**
- Add more tests to the TimeParser class
- Parsing timing is very bug-prone, hence more tests are
needed to ensure the TimeParser API works as intended

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/TimeParserTest.java [hunk 1]
```diff
@@ -37,6 +37,97 @@ void testInvalidDate() {
         assertTrue(hasError);
     }
 
+    @Test
+    void testDayAnd24hTime() throws ParseException {
+        TimeParser.parseDate("Tue 1630");
```
Comment (ghost) [inline src/test/java/seedu/address/logic/parser/TimeParserTest.java]: Perhaps the test method name here could be something along the lines of `parseDate_validDayAnd24hTime_successfulParse`

**Review by ghost — COMMENTED** [review]
Great effort with increasing the test coverage to be more comprehensive.
Just to point out a coding standard that might have been missed in the code:
Test method names should try to follow the three part format `featureUnderTest_testScenario_expectedBehavior()`
with the second and third parts being optional depending on what is being tested exactly 
(parseDate() would contain all the tests for the parseDate method, parseDate_invalidInput() would contain all the test cases with invalid inputs for the parseDate method)

File: src/test/java/seedu/address/logic/parser/TimeParserTest.java (hunk 1 above)
Comment (kiwibang) [reply src/test/java/seedu/address/logic/parser/TimeParserTest.java]: Thanks for the heads up! have edited the names of the tests that I am intending to keep! 👍

**Review by Jonyxzx — APPROVED** [review]
LGTM!

**Pull request merged**